Sources/Overlays/_Testing_Foundation/Attachments/EncodingFormat.swift

```diff
@@ -0,0 +1,84 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+#if canImport(Foundation)
+@_spi(Experimental) import Testing
+import Foundation
+
+#if SWT_TARGET_OS_APPLE && canImport(UniformTypeIdentifiers)
+private import UniformTypeIdentifiers
+#endif
+
+/// An enumeration describing the encoding formats we support for `Encodable`
+/// and `NSSecureCoding` types that conform to `Test.Attachable`.
+enum EncodingFormat {
+  /// A property list format.
+  ///
+  /// - Parameters:
+  ///   - format: The corresponding property list format.
+  case propertyListFormat(_ format: PropertyListSerialization.PropertyListFormat)
+
+  /// The JSON format.
+  case json
+
+  /// The encoding format to use by default.
+  ///
+  /// The specific format this case corresponds to depends on if we are encoding
+  /// an `Encodable` value or an `NSSecureCoding` value.
+  case `default`
```

Comment:
Should this be the first case in the list?

Reply:
I can move it if it bothers you—it's an internal type so it's not going to affect client code.